assets/javascripts/discourse/initializers/add-upcoming-events-to-sidebar.js

```diff
@@ -0,0 +1,40 @@
+import { withPluginApi } from "discourse/lib/plugin-api";
+import I18n from "discourse-i18n";
+
+export default {
+  name: "add-upcoming-events-to-sidebar",
+
+  initialize(container) {
+    const siteSettings = container.lookup("service:site-settings");
+    if (
+      siteSettings.discourse_post_event_enabled &&
+      siteSettings.sidebar_show_upcoming_events
+    ) {
+      withPluginApi("0.8.7", (api) => {
+        api.addCommunitySectionLink((baseSectionLink) => {
+          return class UpcomingEventsSectionLink extends baseSectionLink {
+            get name() {
+              return "upcoming-events";
+            }
+
+            get route() {
+              return "discourse-post-event-upcoming-events";
+            }
+
+            get text() {
+              return I18n.t(
+                "discourse_calendar.discourse_post_event.upcoming_events.title"
+              );
+            }
+
+            get title() {
+              return I18n.t(
+                "discourse_calendar.discourse_post_event.upcoming_events.title"
+              );
+            }
```

Comment:
I know we haven't been using it that way but I don't think there's nothing stopping us from making this more concise.

```suggestion
            name = "upcoming-events";
            route = "discourse-post-event-upcoming-events";
            text = I18n.t(
              "discourse_calendar.discourse_post_event.upcoming_events.title"
            );
            title = I18n.t(
              "discourse_calendar.discourse_post_event.upcoming_events.title"
            );
```

(no need to change it here, just wondering about the api of sidebar links 😃)